### PR TITLE
Support pre-configured switch stack

### PR DESCRIPTION
**Pull request by logankimmel**
Added logic to not count ports that are on another stack as part
of the tear-down

Added tests cases for #check_for_interface

### CODE REVIEW

**Comment by logankimmel** [comment]
@gavin-scott This encompasses your change as I saw it wasn't merged yet. I noticed the tests weren't being run before because of the the name.  That has been fixed too

**Comment by logankimmel** [comment]
@ripienaar Can you review this?  Thanks

**Comment by ripienaar** [comment]
@logankimmel code seems fine, but I cant really comment on if its the right thing to do or not, you mention its based on some changes @gavin-scott made so maybe he is best he does?

File: spec/unit/puppet/provider/force10_interface/dell_ftos_spec.rb [hunk 1]
```diff
@@ -29,7 +29,41 @@
...
+  describe '#exists?' do
+    context 'when vlan exists' do
+      it 'returns true' do
+        vlan_data = eval(File.read(File.join(@fixture_dir,'vlan_information_s')))
+        vlan_info = JSON.parse(vlan_data)
+        provider.stub(:get_vlan_info).and_return(vlan_info)
+        provider.stub(:get_iface).and_return('Tengigabitethernet0/3')
+        expect(provider.exists?).to eq(true)
+      end
+      it 'returns true' do
```
Comment (gavin-scott) [inline spec/unit/puppet/provider/force10_interface/dell_ftos_spec.rb]: Can you update the tests to have unique names? Also please include a line of spacing between the different tests.

File: spec/unit/puppet/provider/force10_interface/dell_ftos_spec.rb [hunk 2]
```diff
@@ -29,7 +29,41 @@
...
+    end
+    context 'when vlan does not exist' do
+      it 'returns false' do
+        vlan_data = eval(File.read(File.join(@fixture_dir,'vlan_information_s')))
+        vlan_info = JSON.parse(vlan_data)
+        provider.stub(:get_vlan_info).and_return(vlan_info)
+        provider.stub(:get_iface).and_return('Tengigabitethernet0/1')
+        expect(provider.exists?).to eq(false)
+      end
+      it 'returns false' do
```
Comment (gavin-scott) [inline spec/unit/puppet/provider/force10_interface/dell_ftos_spec.rb]: Same thing re unique names and spacing.

**Comment by gavin-scott** [comment]
A couple of nits on spacing (would like spaces between all the test blocks) and test names, but looks like this does what I was looking for. Thanks!

**Comment by logankimmel** [comment]
@gavin-scott done

**Pull request merged**